src/reviver.ts

```diff
@@ -35,16 +35,18 @@ export class SafeReviver {
       throw new Error(`Expected key (${key}) to be of type 'string', but got '${typeof(key)}'`);
     }
 
-    // . | used in resource fqn which servers as a key (e.g io.k8s.apimachinery.pkg.apis.meta.v1.APIGroup)
-    // / | used in $ref to point to a definition (e.g #/definitions/io.k8s.apimachinery.pkg.apis.meta.v1.GroupVersionForDiscovery)
-    // - | used in annotation keys (e.g x-kubernetes-group-version-kind)
-    // # | used in $ref to point to a definition (e.g #/definitions/io.k8s.apimachinery.pkg.apis.meta.v1.GroupVersionForDiscovery)
-    // , | used in values that represent a list (e.g merge,retainKeys)
-    const legal = /^(\w|\.|\/|-|#|,)*$/;
+    // .       | used in resource fqn which servers as a key (e.g io.k8s.apimachinery.pkg.apis.meta.v1.APIGroup)
+    // /       | used in $ref to point to a definition (e.g #/definitions/io.k8s.apimachinery.pkg.apis.meta.v1.GroupVersionForDiscovery)
+    // -       | used in annotation keys (e.g x-kubernetes-group-version-kind)
+    // #       | used in $ref to point to a definition (e.g #/definitions/io.k8s.apimachinery.pkg.apis.meta.v1.GroupVersionForDiscovery)
+    // ,       | used in values that represent a list (e.g merge,retainKeys)
+    // <space> | used in enum values with spaces
+    const legalKey = /^(\w|\.|\/|-|#|,)*$/;
+    const legalValue = /^(\w| |\.|\/|-|#|,)*$/;
```

Comment:
I don't like allowing spaces in values in general. Let's scope this down to only allow spaces in `enum` values? Similarly to how we do with `description`. 

https://github.com/cdk8s-team/cdk8s-cli/blob/66c14eb8800b49645247937b9e9d62a899aa8b71/src/reviver.ts#L21

Reply:
I don't think this is possible with our current reviver implementation - `JSON.parse()` only provides key-value pairs to our reviver, and it's invoked on the keys in "inside-out" order -- e.g. for an object like:

```
{ "enum": ["a", "b", "c"] }
```

then the reviver will get called with ("0", "a"), ("1", "b"), ("2", "c"), ("enum", ["a", "b", "c"]). This limits how we can sanitize values since the inner value must be left un-stripped.

I've updated the implementation to narrow the scope of the spaces-specific regex further to only be applied to items in an array (when those values are strings). Let me know if you think this is OK, and/or if I've overlooked some simpler way to go about this.